dev/make-distribution.sh

```diff
@@ -170,24 +170,28 @@ BUILD_COMMAND=("$MVN" clean package -DskipTests $@)
 
 # Actually build the jar
 echo -e "\nBuilding with..."
-echo -e "\$ ${BUILD_COMMAND[@]}\n"
+echo -e "\$ ${BUILD_COMMAND[*]}\n"
 
 "${BUILD_COMMAND[@]}"
 
 # Make directories
 rm -rf "$DISTDIR"
 mkdir -p "$DISTDIR/jars"
 echo "Spark $VERSION$GITREVSTRING built for Hadoop $SPARK_HADOOP_VERSION" > "$DISTDIR/RELEASE"
-echo "Build flags: $@" >> "$DISTDIR/RELEASE"
+echo "Build flags: $*" >> "$DISTDIR/RELEASE"
 
 # Copy jars
 cp "$SPARK_HOME"/assembly/target/scala*/jars/* "$DISTDIR/jars/"
 
 # Only create the yarn directory if the yarn artifacts were built.
-if [ -f "$SPARK_HOME"/common/network-yarn/target/scala*/spark-*-yarn-shuffle.jar ]; then
-  mkdir "$DISTDIR/yarn"
-  cp "$SPARK_HOME"/common/network-yarn/target/scala*/spark-*-yarn-shuffle.jar "$DISTDIR/yarn"
-fi
+for file in "$SPARK_HOME"/common/network-yarn/target/scala*/spark-*-yarn-shuffle.jar
```

Comment:
SC2144 (error): -f doesn't work with globs. Use a for loop.

Reply:
For the same reason as in https://github.com/koalaman/shellcheck/issues/1217 I think we want to suppress this. Looping doesn't make sense here anyway, even if there were multiple files matching

Reply:
I think it will fail after this change too, if there are multiple matching files; looping isn't the intent here. We could maybe check if there is exactly one match? that's the 'right' fix. If there is 1 file, I suppose this logic is the same so it's OK, but it's also a bit wrong in the same way

Reply:
Maybe we can use find to check file count first, then if there's only one file, use cp.